src/plugins/data/common/search/aggs/buckets/range_key.ts

```diff
@@ -19,20 +19,41 @@
 
 const id = Symbol('id');
 
+type Ranges = Array<{
+  from: string | number | undefined;
+  to: string | number | undefined;
+  label?: string;
+}>;
+
 export class RangeKey {
   [id]: string;
   gte: string | number;
   lt: string | number;
+  label?: string;
+
+  private static findCustomLabel(from: string | number, to: string | number, ranges: Ranges) {
+    return ranges
+      ? ranges.find(
+          (range) =>
+            ((from === -Infinity && range.from === undefined) || range.from === from) &&
+            ((to === Infinity && range.to === undefined) || range.to === to)
+        )?.label
+      : undefined;
+  }
 
-  constructor(bucket: any) {
+  constructor(bucket: any, allRanges?: Ranges) {
     this.gte = bucket.from == null ? -Infinity : bucket.from;
     this.lt = bucket.to == null ? +Infinity : bucket.to;
+    this.label = allRanges ? RangeKey.findCustomLabel(this.gte, this.lt, allRanges) : undefined;
```

Comment:
Passing `bucket.from/to` here can simplify the code in the `findCustomLabel` a bit (removing non-finite check).
They effectively equivalent, but the current version is more verbose.

Reply:
Also the ternary operator can be dropped as the missing `ranges` would be handled by `findCustomLabel`

Reply:
Good catch for the ternary operator, I'm not sure whether it makes it much easier because we still have to look for `null`/`undefined`.